### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by crisgarner**
- Deploy scripts for DAI, USDC, WBTC, and ETH hard mode vaults
- Added minimum TCAP.

### CODE REVIEW

File: deploy/mainnet/005_hard_vaults.ts [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+					ratio,
+					burnFee,
+					liquidationPenalty,
+					tcapOracle,
+					tcapAddress,
+					WETHContract,
+					ethOracle,
+					ethOracle,
+					treasury,
+					0
```
Comment (voith) [inline deploy/mainnet/005_hard_vaults.ts]: Why is minimum TCAP 0? should it be 20 * 10 ** 18?

File: deploy/mainnet/005_hard_vaults.ts [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+					ratio,
+					burnFee,
+					liquidationPenalty,
+					tcapOracle,
+					tcapAddress,
+					DAIContract,
+					daiOracle,
+					ethOracle,
+					treasury,
+					0
```
Comment (voith) [inline deploy/mainnet/005_hard_vaults.ts]: Why is minimum TCAP 0? should it be 20 * 10 ** 18?

File: deploy/mainnet/005_hard_vaults.ts [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+					ratio,
+					burnFee,
+					liquidationPenalty,
+					tcapOracle,
+					tcapAddress,
+					USDCContract,
+					usdcOracle,
+					ethOracle,
+					treasury,
+					0
```
Comment (voith) [inline deploy/mainnet/005_hard_vaults.ts]: Why is minimum TCAP 0? should it be 20 * 10 ** 18?

**Review by voith — COMMENTED** [review]
Can you please check the "\t" and the burn Fee?

File: deploy/mainnet/005_hard_vaults.ts (hunk 1 above)
Comment (voith) [reply deploy/mainnet/005_hard_vaults.ts]: resolved in df294e2deaaf4db088fe0aa6a2f50014e43970f7

File: deploy/mainnet/005_hard_vaults.ts (hunk 2 above)
Comment (voith) [reply deploy/mainnet/005_hard_vaults.ts]: resolved in https://github.com/cryptexfinance/contracts/commit/df294e2deaaf4db088fe0aa6a2f50014e43970f7

File: deploy/mainnet/004_usdc_collateral.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+import { ethers } from "ethers";
+
+const usdcVaultHandler: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
+	if (hardhatArguments.network === "mainnet") {
+		console.log("=====Mainnet Deploy=====");
+		const { log } = deployments;
+
+		const namedAccounts = await hre.getNamedAccounts();
+		let usdcOracle = await deployments.getOrNull("UsdcOracle");
+		const usdcAggregator = "\t0x8fFfFfd4AfB6115b954Bd326cbe7B4BA576818f6";
```
[the remark this answers is not in the document]
Comment (crisgarner) [reply deploy/mainnet/004_usdc_collateral.ts:14]: it's not, might gotten copied from coingecko

File: deploy/mainnet/005_hard_vaults.ts [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+	if (hardhatArguments.network === "mainnet") {
+		console.log("=====Hardmode Mainnet Deploy=====");
+		const { log } = deployments;
+
+		const hardETHVaultHandler = await deployments.getOrNull("HardETHVaultHandler");
+		const hardDAIVaultHandler = await deployments.getOrNull("HardDAIVaultHandler");
+		const hardUSDCVaultHandler = await deployments.getOrNull("HardUSDCVaultHandler");
+
+		const namedAccounts = await hre.getNamedAccounts();
+		const orchestrator = "0x373c74bce7893097ab26d22f05691907d4f2c18e";
```
[the remark this answers is not in the document]
Comment (voith) [reply deploy/mainnet/005_hard_vaults.ts:18]: cool

File: scripts/CIP-TOWER.ts [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+import {BigNumber} from "ethers";
+
+async function main() {
+	const ethers = hre.ethers;
+
+	let orchestratorAddress = "0x373C74BcE7893097ab26d22f05691907D4f2c18e";
+	let tcap = await deployments.get("TCAP");
+	//TODO: waiting for real deployment
+	let tcapContract = await ethers.getContractAt("TCAP", tcap.address);
+	let usdcHardVault = await deployments.get("AaveVaultHandler");
```
Comment (voith) [inline scripts/CIP-TOWER.ts:14]: If we are going with hardcoded values, can we use hardcoded values here?

**Review by voith — APPROVED** [review]
LGTM.
Just one question about the hardcoded values.

File: scripts/CIP-TOWER.ts (hunk 6 above)
Comment (crisgarner) [reply scripts/CIP-TOWER.ts:14]: Yes! left it like that for now as contracts are not deployed

**Pull request merged**